File: javascript/Algorithm3.js
// Algorithm 3: Find Roots of a Quadratic Equation ax2 + bx + c = 0
const readline = require('readline');
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});
rl.question('Enter value of a: ', (a) => {
    rl.question('Enter value of b: ', (b) => {
        rl.question('Enter value of c: ', (c) => {
            a = parseFloat(a);
            b = parseFloat(b);
            c = parseFloat(c);
            let D = b * b - 4 * a * c;
            if (D >= 0) {
                let r1 = (-b + Math.sqrt(D)) / (2 * a);
                let r2 = (-b - Math.sqrt(D)) / (2 * a);
                console.log(`The roots are: ${r1} and ${r2}`);
            } else {
                let rp = -b / (2 * a);
                let ip = Math.sqrt(-D) / (2 * a);
                console.log(`The roots are: ${rp} + j${ip} and ${rp} - j${ip}`);
            }
            rl.close();
        });
    });
});

File: python/Algorithm3.py
# Algorithm 3: Find Roots of a Quadratic Equation ax2 + bx + c = 0
import math
a = int(input("Enter value of a: "))
b = int(input("Enter value of b: "))
c = int(input("Enter value of c: "))
D = (b*b) - 4*a*c
if D >= 0:
  r1 = (-b + math.sqrt(D))/(2*a)
  r2 = (-b - math.sqrt(D))/(2*a)
  print("The roots are: ", r1, "and", r2)
else:
  rp = -b/(2*a)
  ip = math.sqrt(-D)/(2*a)
  print("The roots are: ", rp, "+j", ip, "and", rp, "-j", ip)